### PR TITLE
Feat(eos_cli_config_gen): Add support for permit response traffic nat under ip acls

### LINKED ISSUE #4119: Feat(eos_cli_config_gen): Add support for "permit response traffic nat" under ip acls
Add support for "permit response traffic nat" under ip acls in eos_cli_config_gen. It should be added as a string only accepting `"nat"` for now. Add it at the same level as counter option. The cli should be printed after all sequences have been printed.

The schema doc should mention minimum EOS 4.32.2F

```yaml
permit_response_traffic : <str; "nat">
```

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Add support for permit response traffic nat under ip acls

## Related Issue(s)

Fixes #4119 

## Component(s) name

`arista.avd.eos_cli_config_gen`

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/acl.md [hunk 1]
```diff
@@ -140,6 +140,8 @@ ACL has counting mode `counters per-entry` enabled!
 | 20 | permit ip 10.0.0.0/8 any |
 | 30 | permit ip 192.0.2.0/24 any |
 
+ACL has permit response traffic rule `permit response traffic nat` enabled!
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/acl.md]: Please add it in the table like `| - |  permit response traffic nat |` instead. Thanks.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/access_lists.schema.yml [hunk 2]
```diff
@@ -21,6 +21,13 @@ keys:
           convert_types: [ int ]
         counters_per_entry:
           type: bool
+        permit_response_traffic:
+          type: str
+          valid_values:
+            - nat
+          description: |-
+            NAT based reflexive ACL.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/access_lists.schema.yml]:
```suggestion
            Permit response traffic automatically based on NAT translations.
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/acl.md (hunk 1 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/acl.md]: Done!

File: python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/access-lists.j2 [hunk 3]
```diff
@@ -22,6 +22,9 @@ ACL has counting mode `counters per-entry` enabled!
 {%         for sequence in access_list.sequence_numbers | arista.avd.natural_sort('sequence') %}
 | {{ sequence.sequence }} | {{ sequence.action }} |
 {%         endfor %}
+{%         if access_list.permit_response_traffic is arista.avd.defined %}
+| - | permit response traffic nat |
```
Comment (ayushmittal-arista) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/access-lists.j2:26]:
```suggestion
| - | permit response traffic nat |
```
| - | permit response traffic {{ access_list.permit_response_traffic }} |

Comment (ayushmittal-arista) [inline python-avd/pyavd/_eos_cli_config_gen/j2templates/documentation/access-lists.j2]:
```suggestion
| - | permit response traffic {{ access_list.permit_response_traffic }} |
```

**Review by ClausHolbechArista — APPROVED** [review]

**Review by Shivani-gslab — APPROVED** [review]
LGTM!

**Review by gmuloc — APPROVED** [review]

**Pull request merged**